### PR TITLE
[staging] pythonPackages.lxml: 4.3.3 -> 4.4.1, reduce closure size

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
was reducing the closure size and noticed it was out of date.

Looks like the dependencies were being passed through propagatedBuildInputs, which was causing a large closure bloat.

```
$ nix path-info -Sh ./result
/nix/store/mmfq7029d9ziawpl02grxz4jjfrklhwn-python3.7-lxml-4.3.3         160.5M
$ nix path-info -Sh ./result
/nix/store/4ig7nll10z63liiwbyhcmsy9c8yizmfz-python3.7-lxml-4.4.1         103.1M
```

roughly a 57MB savings :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

EDIT: and add myself as maintainer

### CODE REVIEW

File: pkgs/development/python-modules/lxml/default.nix [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
 
-  meta = {
+  pythonImportsCheck = [ "lxml" "lxml.etree" ];
+
+  meta = with stdenv.lib; {
     description = "Pythonic binding for the libxml2 and libxslt libraries";
-    homepage = https://lxml.de;
-    license = stdenv.lib.licenses.bsd3;
-    maintainers = with stdenv.lib.maintainers; [ sjourdois ];
+    homepage = "https://lxml.de";
```
Comment (FRidh) [inline pkgs/development/python-modules/lxml/default.nix]: please undo this change, urls are their own type.

**Comment by FRidh** [comment]
That's a really big improvement!

File: pkgs/development/python-modules/lxml/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/lxml/default.nix]: I thought there was a general consensus for moving away from raw url's: https://github.com/NixOS/rfcs/pull/45

Comment (FRidh) [reply pkgs/development/python-modules/lxml/default.nix]: It's not been accepted yet.

**Comment by jonringer** [comment]
sorry for request spam, had to change base, as this rebuilds >1000 packages

**Comment by FRidh** [comment]
hydra job https://hydra.nixos.org/eval/1549994

**Comment by jonringer** [comment]
oh thanks, for kicking that off. Should I close this, and make a python-unstable -> staging PR, and just fix build errors in that PR?

File: pkgs/development/python-modules/lxml/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/lxml/default.nix]: that's fair, I'll revert

Comment (jonringer) [reply pkgs/development/python-modules/lxml/default.nix]: revereted

**Comment by jonringer** [comment]
@FRidh other than some timeouts, it looks like it was able to eval without any regression, merge?
